flutter/Graminsta/lib/main.dart

```diff
@@ -41,21 +48,57 @@ class _HomePageState extends State<HomePage> {
     });
   }
 
+  void _logout() async {
+    //clear SharedPreferences and redirect to AuthPage.
+    final result = await http.clearAuthToken();
+    if (result) {
+      Navigator.of(context).pushNamedAndRemoveUntil(
+        '/auth',
+        (route) => route == null,
+      );
+    }
+  }
+
+  getUserToken() async {
+    final prefs = await SharedPreferences.getInstance();
+    final token = prefs.getString('user_token') ?? '';
+    debugPrint('user_token: $token');
+    return token;
+  }
+
   @override
   Widget build(BuildContext context) {
+    //check user login status. If there is no token, redirect to AuthPage.
+    getUserToken().then((token) => {
+          if (token == "")
+            {
+              Navigator.of(context).pushNamedAndRemoveUntil(
+                '/auth',
+                (route) => route == null,
+              )
+            }
+        });
```

Comment:
Use FutureBuilder instead, since this process is `async`. But I'd suggest you do that in another PR to keep this PR small.

See also: https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html

Reply:
I will open another PR to fix this.